pyvptree/src/PythonBindings.cpp

```diff
@@ -23,14 +24,18 @@ template <distance_func_f distance> class VPTreeNumpyAdapter {
     public:
     VPTreeNumpyAdapter() = default;
 
-    void set(const ndarrayf &array) { tree = vptree::VPTree<arrayf, float, distance>(array); }
+    void set(const ndarrayf &array) { tree.set(array); }
 
-    std::tuple<std::vector<std::vector<unsigned int>>, std::vector<std::vector<float>>> searchKNN(const ndarrayf &queries, unsigned int k) {
+    std::tuple<std::vector<std::vector<int64_t>>, std::vector<std::vector<float>>> searchKNN(const ndarrayf &queries, unsigned int k) {
+
+        if (tree.isEmpty()) {
+            throw std::runtime_error("indices must be first initialized with .set function");
+        }
```

Comment:
should this check be in the searchKNN implementation of the tree?

Reply:
Moved all checks to VPTree, indeed its better since it works in C++ as well.
Fixed on 15699fdf8f0723e32cc5c67ba4d2b00cc0aea894